shared/src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp

```diff
@@ -228,6 +228,71 @@ namespace datadog::shared::nativeloader
                             "but an unsupported command was detected");
                         return CORPROF_E_PROFILER_CANCEL_ACTIVATION;
                     }
+
+                    // Additional Special case CI Visibility checks
+                    bool is_ci_visibility_enabled = false;
+                    if (TryParseBooleanEnvironmentValue(GetEnvironmentValue(EnvironmentVariables::CiVisibilityEnabled), is_ci_visibility_enabled)
+                        && is_ci_visibility_enabled)
+                    {
+                        if (tokenized_command_line[1] != WStr("test") &&
+                            process_command_line.find(WStr("testhost")) == WSTRING::npos &&
+                            process_command_line.find(WStr("exec")) == WSTRING::npos &&
+                            process_command_line.find(WStr("datacollector")) == WSTRING::npos &&
+                            process_command_line.find(WStr("vstest.console.dll")) == WSTRING::npos)
+                        {
+                            Log::Info("The Tracer Profiler has been disabled because the process is running in CI Visibility "
+                                "mode, the name is 'dotnet' but the commandline doesn't contain 'testhost' or 'datacollector' or 'vstest.console.dll' or 'exec'");
+                            return CORPROF_E_PROFILER_CANCEL_ACTIVATION;
+                        }
+                    }
+                }
+            }
+        }
+
+        // AAS checks
+        const auto isRunningInAas = TryParseBooleanEnvironmentValue(
```

Comment:
We should be careful with `DD_AZURE_APP_SERVICES`. It doesn't mean "we're running inside AAS," it means "we're instrumenting _with the site extension_."

We now have a nuget package that can be used instead of the site extension, and `DD_AZURE_APP_SERVICES` won't be set in that case.

Reply:
(no action required in this PR, just an observation)